x-pack/plugins/security/server/usage_collector/security_usage_collector.test.ts

```diff
@@ -0,0 +1,425 @@
+      ?.fetch(clusterClient.asScoped().callAsCurrentUser);
+
+    expect(usage).toEqual({
+      auditLoggingEnabled: false,
+      accessAgreementEnabled: false,
+      authProviderCount: 1,
+      enabledAuthProviders: ['saml'],
+      loginSelectorEnabled: true,
+      httpAuthSchemes: ['apikey'],
+    });
+  });
+
+  it('reports when audit logging is enabled', async () => {
+    const config = createSecurityConfig(
+      ConfigSchema.validate({
+        audit: {
+          enabled: true,
+        },
+      })
+    );
+    const usageCollection = usageCollectionPluginMock.createSetupContract();
+    const license = createSecurityLicense({ isLicenseAvailable: true, allowAuditLogging: true });
+    registerSecurityUsageCollector({ usageCollection, config, license });
+
+    const usage = await usageCollection
+      .getCollectorByType('security')
+      ?.fetch(clusterClient.asScoped().callAsCurrentUser);
+
+    expect(usage).toEqual({
+      auditLoggingEnabled: true,
+      accessAgreementEnabled: false,
+      authProviderCount: 1,
+      enabledAuthProviders: ['basic'],
+      loginSelectorEnabled: false,
+      httpAuthSchemes: ['apikey'],
+    });
+  });
+
+  it('does not report audit logging when the license does not permit it', async () => {
+    const config = createSecurityConfig(
+      ConfigSchema.validate({
+        audit: {
+          enabled: true,
+        },
+      })
+    );
+    const usageCollection = usageCollectionPluginMock.createSetupContract();
+    const license = createSecurityLicense({ isLicenseAvailable: true, allowAuditLogging: false });
+    registerSecurityUsageCollector({ usageCollection, config, license });
+
+    const usage = await usageCollection
+      .getCollectorByType('security')
+      ?.fetch(clusterClient.asScoped().callAsCurrentUser);
+
+    expect(usage).toEqual({
+      auditLoggingEnabled: false,
+      accessAgreementEnabled: false,
+      authProviderCount: 1,
+      enabledAuthProviders: ['basic'],
+      loginSelectorEnabled: false,
+      httpAuthSchemes: ['apikey'],
+    });
+  });
+
+  it('reports any customized http auth schemes', async () => {
+    const config = createSecurityConfig(
+      ConfigSchema.validate({
+        authc: {
+          http: {
+            schemes: ['custom1', 'custom2'],
+          },
+        },
+      })
+    );
+    const usageCollection = usageCollectionPluginMock.createSetupContract();
+    const license = createSecurityLicense({ isLicenseAvailable: true, allowAuditLogging: false });
+    registerSecurityUsageCollector({ usageCollection, config, license });
+
+    const usage = await usageCollection
+      .getCollectorByType('security')
+      ?.fetch(clusterClient.asScoped().callAsCurrentUser);
+
+    expect(usage).toEqual({
+      auditLoggingEnabled: false,
+      accessAgreementEnabled: false,
+      authProviderCount: 1,
+      enabledAuthProviders: ['basic'],
+      loginSelectorEnabled: false,
+      httpAuthSchemes: ['custom1', 'custom2'],
```

Comment:
This is the only metric that collects custom strings. Is there any potential for a privacy risk here?

Reply:
I gave this some thought when I first put this together, and I didn't foresee a privacy risk. The [list of available auth schemes](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml) doesn't seem like it'd divulge anything we shouldn't collect. That being said, IANAL. I added this at @azasypkin's request, but if we're at all concerned, then I don't mind pulling it or revisiting this in a followup.

Reply:
Ah, so if I'm understanding correctly, users probably won't be using non-standard auth schemes?

Perhaps we could simply check against that list and redact any that don't match, just in case someone's doing something truly custom?

Reply:
> Ah, so if I'm understanding correctly, users probably won't be using non-standard auth schemes?

More than likely not, as I'd expect this would require a custom auth provider in Elasticsearch, or between ES and Kibana.

> Perhaps we could simply check against that list and redact any that don't match, just in case someone's doing something truly custom?

Yeah, that sounds reasonable to me.